pkg/v1/tarball/layer.go

```diff
@@ -249,15 +250,17 @@ func LayerFromOpener(opener Opener, opts ...LayerOption) (v1.Layer, error) {
 }
 
 // LayerFromReader returns a v1.Layer given a io.Reader.
+//
+// The reader's contents are read and buffered to a temp file in the process.
 func LayerFromReader(reader io.Reader, opts ...LayerOption) (v1.Layer, error) {
-	// Buffering due to Opener requiring multiple calls.
-	a, err := ioutil.ReadAll(reader)
+	tmp, err := ioutil.TempFile("", "")
```

Comment:
Are we okay with leaking this file?

Reply:
I feel like this is just a misleading interface and we should make people use `LayerFromFile` or `pkg/v1/stream`.

Reply:
I'd be totally fine with marking it deprecated, moving it to `internal` and only using it for testing.

We only use it ourself for [`crane.Layer(map[string][]byte)`](https://pkg.go.dev/github.com/google/go-containerregistry/pkg/crane#Layer) (discussion in https://github.com/google/go-containerregistry/pull/485), and there's no real reason that needs to use this.

Should we just deprecate it and recommend against its usage, favoring `LayerFromFile`/`LayerFromOpener` instead?

Reply:
Yeah I think that's the easiest thing to do for now. `crane.Layer` is trivially fixable to use FromOpener.

We can still keep this change but also add Deprecated? Better to leak a temp file than buffer, maybe?